library/src/main/java/com/google/android/material/motion/runtime/TargetScope.java

```diff
@@ -118,6 +134,23 @@ private int getDetailedState() {
     return state;
   }
 
+  private void removeNamedPlan(PlanInfo planInfo, Performer performer) {
+    if (performer != null) {
+      if (performer instanceof Performer.NamedPlanPerformance) {
+        ((Performer.NamedPlanPerformance) performer).removePlan((NamedPlan) planInfo.plan, planInfo.name);
+      }
+      namedCache.remove(planInfo.name);
+    }
+  }
+
+  private Performer getNamedPerformer(PlanInfo planInfo) {
+    Performer namedPerformer = namedCache.get(planInfo.name);
+    if (namedPerformer == null) {
+      namedPerformer = getPerformer(planInfo);
```

Comment:
Yes - this should not delegate to `getPerformer()`. You should be able to remove this entire method and just inline `namedCache.get(name)`.